lib/shopify_app/login_protection.rb

```diff
@@ -41,20 +41,25 @@ def redirect_to_login
         head :unauthorized
       else
         session[:return_to] = request.fullpath if request.get?
-        redirect_to main_or_engine_login_url(shop: params[:shop])
+        redirect_to login_url
       end
     end
 
     def close_session
       session[:shopify] = nil
       session[:shopify_domain] = nil
-      redirect_to main_or_engine_login_url(shop: params[:shop])
+      redirect_to login_url
     end
 
-    def main_or_engine_login_url(params = {})
-      main_app.login_url(params)
-    rescue NoMethodError
-      shopify_app.login_url(params)
+    def login_url
+      url = ShopifyApp.configuration.login_url
+
+      if params[:shop].present?
+        query = { shop: params[:shop] }.to_query
+        url = "#{url}?#{query}"
```

Comment:
I see! We set that [here](https://github.com/Shopify/shopify_app/blob/7a654313848c7c0a5e0ceceacde1969a1400b03a/lib/shopify_app/login_protection.rb#L43)